actors/builtin/market/market_test.go

```diff
@@ -1 +1,252 @@
 package market_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+	"github.com/filecoin-project/specs-actors/actors/abi/big"
+	"github.com/filecoin-project/specs-actors/actors/builtin"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/filecoin-project/specs-actors/actors/builtin/miner"
+	"github.com/filecoin-project/specs-actors/actors/runtime/exitcode"
+	"github.com/filecoin-project/specs-actors/actors/util/adt"
+	"github.com/filecoin-project/specs-actors/support/mock"
+	tutil "github.com/filecoin-project/specs-actors/support/testing"
+)
+
+func TestActorConstruction(t *testing.T) {
+	actor := market.Actor{}
+	receiver := tutil.NewIDAddr(t, 100)
+	builder := mock.NewBuilder(context.Background(), receiver).
+		WithCaller(builtin.SystemActorAddr, builtin.InitActorCodeID)
+
+	t.Run("simple construction", func(t *testing.T) {
+		rt := builder.Build(t)
+
+		rt.ExpectValidateCallerAddr(builtin.SystemActorAddr)
+
+		params := adt.EmptyValue{}
+		ret := rt.Call(actor.Constructor, &params).(*adt.EmptyValue)
+
+		assert.Equal(t, adt.EmptyValue{}, *ret)
+		rt.Verify()
```

Comment:
Move the constructor state checks out from the state test into this one (and delete the state test)